### PR TITLE
Make History and Memory list items context menu accessible using keyboard

### LINKED ISSUE #311: "History" list item context menu is not accessible using keyboard
**Repro Steps:**
* Launch Calculator app.
* Navigate to History list item and try to open the context menu
* Observe the issue.

**Actual Result**
History list item context menu is not accessible using keyboard. It is accessible using mouse.

**Expected Result**
History list item context menu should be accessible using keyboard.

App Version: 10.1902.16.0

### PR DESCRIPTION
**Pull request by rudyhuyn**
### Description of the changes:
- Move History and Memory context menus to ListViewItemContainer so users can open them with keyboards
- Modify how Memory manages context menus to use a similar approach than History (a lot simplier)

### How changes were validated:
- Manually test using keyboard, mouse and touch.

### CODE REVIEW

File: src/Calculator/Views/HistoryList.xaml.cpp [hunk 1]
```diff
@@ -67,9 +67,13 @@ void HistoryList::ListView_ItemClick(_In_ Object^ sender, _In_ ItemClickEventArg
 
 void HistoryList::OnDeleteMenuItemClicked(_In_ Object^ sender, _In_ RoutedEventArgs^ e)
 {
-    auto clickedItem = safe_cast<HistoryItemViewModel^>(safe_cast<FrameworkElement^>(sender)->DataContext);
+    auto listViewItem = HistoryContextMenu->Target;
+    auto itemViewModel = safe_cast<HistoryItemViewModel^>(HistoryListView->ItemFromContainer(listViewItem));
```
Comment (joseartrivera) [inline src/Calculator/Views/HistoryList.xaml.cpp]: safe_cast would throw here if the cast didn't work, so the below check for nullptr would never occur. I think you want dynamic_cast instead

File: src/Calculator/Views/HistoryList.xaml [hunk 2]
```diff
@@ -51,17 +51,15 @@
...
                                 Click="OnDeleteMenuItemClicked"
                                 Icon="Delete"/>
             </MenuFlyout>
 
             <DataTemplate x:Key="HistoryItemTemplate" x:DataType="model:HistoryItemViewModel">
                 <muxc:SwipeControl RightItems="{StaticResource HistorySwipeItems}">
-                    <StackPanel Margin="0,6,4,6"
-                                Background="Transparent"
-                                ContextFlyout="{StaticResource HistoryContextMenu}">
+                    <StackPanel Margin="0,6,4,6"Background="Transparent">
```
Comment (joseartrivera) [inline src/Calculator/Views/HistoryList.xaml]: Add a space before Background

File: src/Calculator/Views/Memory.xaml.cpp [hunk 3]
```diff
@@ -135,7 +112,6 @@ void Memory::MemoryList_Unloaded(_In_ Object^ sender, _In_ RoutedEventArgs^ e)
 
 MemoryItemViewModel^ Memory::GetMemoryItemForCurrentFlyout()
 {
-    auto listViewItem = m_memoryItemFlyout->Target;
-
+    auto listViewItem = MemoryContextMenu->Target;
     return safe_cast<MemoryItemViewModel^>(MemoryListView->ItemFromContainer(listViewItem));
```
Comment (joseartrivera) [inline src/Calculator/Views/Memory.xaml.cpp]: Same feedback here safe_cast would throw so the checks for nullptr in the calling code would not occur.

File: src/Calculator/Views/HistoryList.xaml.cpp (hunk 1 above)
Comment (rudyhuyn) [reply src/Calculator/Views/HistoryList.xaml.cpp]: Sorry for the delay, I was very busy these past few days, I will modify that tonight!

**Comment by joseartrivera** [comment]
@rudyhuyn this looks good, could you resolve the merge conflicts and then I can merge it in?

**Pull request merged**